mtags/src/main/scala-3/scala/meta/internal/pc/completions/InterpolatorCompletions.scala

```diff
@@ -0,0 +1,249 @@
+          snippetsEnabled
+        )
+    end match
+  end contribute
+
+  /**
+   * Find the identifier that corresponds to the previous interpolation splice.
+   * For string `s" $Main.metho@@ "` we want to get `Main` identifier.
+   * The difference with Scala 2 is that we search for it through the path using
+   * the created partial function.
+   */
+  private def interpolatorMemberArg(
+      lit: Literal,
+      parent: Tree
+  ): PartialFunction[Tree, Option[Ident | Select]] =
+    case tree @ Apply(
+          _,
+          List(Typed(expr: SeqLiteral, _))
+        ) if expr.elems.exists {
+          case _: Ident => true
+          case _: Select => true
+          case _ => false
+        } =>
+      val allLiterals = parent match
+        case SeqLiteral(elems, _) =>
+          elems
+        case _ => Nil
+      expr.elems.zip(allLiterals.tail).collectFirst {
+        case (i: (Ident | Select), literal) if literal == lit =>
+          i
+      }
+  end interpolatorMemberArg
+
+  /**
+   * A completion to select type members inside string interpolators.
+   *
+   * Example: {{{
+   *   // before
+   *   s"Hello $name.len@@!"
+   *   // after
+   *   s"Hello ${name.length()$0}"
+   * }}}
+   */
+  private def contributeMember(
+      lit: Literal,
+      path: List[Tree],
+      text: String,
+      cursor: SourcePosition,
+      completionPos: CompletionPos,
+      completions: Completions,
+      areSnippetsSupported: Boolean
+  )(using Context): List[CompletionValue] =
+    def newText(
+        name: String,
+        suffix: Option[String],
+        identOrSelect: Ident | Select
+    ): String =
+      val snippetCursor = suffixEnding(suffix, areSnippetsSupported)
+      new StringBuilder()
+        .append('{')
+        .append(
+          text.substring(identOrSelect.span.start, identOrSelect.span.end)
+        )
+        .append('.')
+        .append(name.backticked)
+        .append(snippetCursor)
+        .append('}')
+        .toString
+    end newText
+
+    val memberCompletions = for
+      parent <- path.tail.headOption.toList
+      if text.charAt(lit.span.point - 1) != '}'
+      identOrSelect <- path
+        .collectFirst(interpolatorMemberArg(lit, parent))
+        .flatten
+    yield identOrSelect.symbol.info.allMembers.collect {
+      case m
+          if CompletionFuzzy.matches(
+            completionPos.query,
+            m.symbol.name.toString()
+          ) =>
+        val sym = m.symbol
+        val label = sym.name.decoded
+        completions.completionsWithSuffix(
+          sym,
+          label,
+          (name, sym, suffix) =>
+            CompletionValue.Interpolator(
+              sym,
+              label,
+              Some(newText(name, suffix, identOrSelect)),
+              Nil,
+              Some(cursor.withStart(identOrSelect.span.start).toLSP),
+              // Needed for VS Code which will not show the completion otherwise
+              Some(identOrSelect.show + "." + label)
```

Comment:
I changed one controversial thing here, which is needed for VS Code to show the completion. Otherwise it will skip it unfortunately.